src/redis-cli.c

```diff
@@ -1436,15 +1446,16 @@ static int cliSendCommand(int argc, char **argv, long repeat) {
                 cliRefreshPrompt();
             } else if (!strcasecmp(command,"exec") && argc == 1 && config.in_multi) {
                 config.in_multi = 0;
-                if (config.last_cmd_type == REDIS_REPLY_ERROR) {
-                    config.dbnum = config.pre_multi_dbnum;
+                if (config.last_cmd_type == REDIS_REPLY_ERROR ||
+                        config.last_cmd_type == REDIS_REPLY_NIL) {
+                    config.input_dbnum = config.dbnum = config.pre_multi_dbnum;
```

Comment:
styling/consistency, in redis, when we have multi-line `if`, we (usually) move the opening bracket to a line of it's own.
```suggestion
                if (config.last_cmd_type == REDIS_REPLY_ERROR ||
                    config.last_cmd_type == REDIS_REPLY_NIL)
                {
                    config.input_dbnum = config.dbnum = config.pre_multi_dbnum;
```